dask_ml/preprocessing/_block_transformer.py

```diff
@@ -5,47 +5,56 @@
 from dask import compute
 from pandas.api.types import is_categorical_dtype
 from scipy import stats
+from sklearn.base import BaseEstimator, TransformerMixin
 from sklearn.preprocessing import FunctionTransformer
 from sklearn.utils.validation import check_is_fitted, check_random_state
 
 from dask_ml._utils import copy_learned_attributes
 from dask_ml.utils import check_array, handle_zeros_in_scale
 
 
-class BlockTransformer(FunctionTransformer):
-    def __init__(
-        self,
-        func=None,
-        inverse_func=None,
-        validate=None,
-        accept_sparse=False,
-        check_inverse=True,
-        kw_args=None,
-        inv_kw_args=None,
-    ):
-        super(BlockTransformer, self).__init__(
-            func=func,
-            inverse_func=inverse_func,
-            validate=validate,
-            accept_sparse=accept_sparse,
-            check_inverse=check_inverse,
-            kw_args=kw_args,
-            inv_kw_args=inv_kw_args,
-        )
+class BlockTransformer(BaseEstimator, TransformerMixin):
+    """Construct a transformer from a an arbitrary callable
+
+    The BlockTransformer forwards the blocks of the X arguments to a user-defined
+    function or function-object an returns the result of this operation.
+    This is useful for stateless operations, that can be performed on the cell or
+    block level, such as taking the log of frequencies. In general the transformer
+    is not suitable for e.g. standardization tasks as this requires information for
+    a complete columns
+
+    Parameters
+    ----------
+    func : callable
+        The callable to use for the transformation.
+
+    validate : bool, optional default=True
+         Indicate that the input X array should be checked before calling
+        ``func``.
+
+    kw_args : dict, optional
+        Dictionary of additional keyword arguments to pass to func.
+    """
+
+    def __init__(self, func=None, validate=True, kw_args=None):
+        self.func = func
+        self.validate = validate
+        self.accept_sparse = False
+        self.kw_args = kw_args
 
     def fit(self, X):
         return self
 
     def transform(self, X):
+        def func_forwarded(arr):
```

Comment:
defining this function, will use the name 'func_forwared' in the desk computation graph: `dask.array<func_forwarded, shape=(5, 2), dtype=int64, chunksize=(3, 2)>` is there a suggested way to solve this issue here? Probably, it would have been better to wrap the function properly and use the `@wraps` decorator. Right idea?

Reply:
Hmm, something like that.

Can you instead do

```python
kw_args = self.kw_args or {}
# array
XP = X.map_blocks(self.func, dtype=X.dtype, chunks=X.chunks, **kw_args)
# dataframe
...

```

Reply:
I'll try that, and investigate if `map_blocks` passes the arguments correctly to `func`